transformer_engine/pytorch/distributed.py

```diff
@@ -524,6 +524,17 @@ def _is_te_module(module):
     return is_te_module
 
 
+def _has_te_submodule(root_module):
+    """
+    Check if given root module contains any Transformer Engine submodules.
+    """
+    if isinstance(root_module, torch.nn.Module):
+        for submodule in root_module.children():
```

Comment:
@denera I think by using `root_module.modules()` instead, we can do away with recursion as well as requiring 2 functions (has/is module) since it also returns self

Reply:
Didn't realize `torch.nn.Module.modules()` recursed into all submodules in the network. Definitely makes things easier. Fixed in the latest commit.